### PR TITLE
feat: sync local packages from remote repository

### PR DESCRIPTION
**Pull request by OrangeX4**
1. add a config of syncRepo for vscode
2. commands `tinymist.pullSyncRepo` and `tinymist.typstSync` for vscode

I didn't find a simple git library in rust, and the asynchronous cancelable commands to adapt to vscode were complicated, so I didn't have enough energy for the time being.

### CODE REVIEW

File: editors/vscode/src/git-sync.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    location: ProgressLocation.Notification,
+    title: 'Syncing',
+    cancellable: true
+  }, async (progress, token) => {
+    // 0. cancel callback
+    token.onCancellationRequested(() => {
+      window.showInformationMessage('Syncing with syncRepo canceled.');
+    });
+    // 1. get syncRepo git
+    progress.report({ increment: 15, message: 'Getting syncRepo...' });
```
Comment (Myriad-Dreamin) [inline editors/vscode/src/git-sync.ts:78]: We can convert `syncRepo` to natural English.

File: editors/vscode/src/git-sync.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    location: ProgressLocation.Notification,
+    title: 'Pulling',
+    cancellable: true
+  }, async (progress, token) => {
+    // 0. cancel callback
+    token.onCancellationRequested(() => {
+      window.showInformationMessage('Pulling from syncRepo canceled.');
+    });
+    // 1. get syncRepo git
+    progress.report({ increment: 33, message: 'Getting syncRepo...' });
```
Comment (Myriad-Dreamin) [inline editors/vscode/src/git-sync.ts:122]: Here as well.

File: editors/vscode/src/git-sync.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+export async function commandPullRepo() {
+  await window.withProgress({
+    location: ProgressLocation.Notification,
+    title: 'Pulling',
+    cancellable: true
+  }, async (progress, token) => {
+    // 0. cancel callback
+    token.onCancellationRequested(() => {
+      window.showInformationMessage('Pulling from syncRepo canceled.');
```
Comment (Myriad-Dreamin) [inline editors/vscode/src/git-sync.ts:119]: And here.

File: editors/vscode/package.json [hunk 4]
```diff
@@ -568,6 +574,26 @@
...
+                "title": "Create Typst Local Package",
+                "category": "Typst"
+            },
+            {
+                "command": "tinymist.openLocalPackage",
+                "title": "Open Typst Local Package",
+                "category": "Typst"
+            },
+            {
+                "command": "tinymist.pullSyncRepo",
```
Comment (Myriad-Dreamin) [inline editors/vscode/package.json:588]: The name of the command could be bikeshedded. Candidate renaming:
- `pullSyncRepo` -> `pullLocalPackage`.
- `typstSync` -> `syncLocalPackage`.

Additionally, what's different between `pull` and `sync`? Could you describe it for users here?

**Review by Myriad-Dreamin — COMMENTED** [review]
LGTM

**Comment by OrangeX4** [comment]
ready.

**Comment by Myriad-Dreamin** [comment]
Could you rebase it on to main?

**Comment by Myriad-Dreamin** [comment]
close it since it probably contains design defect.

**Pull request closed without merging**